### PR TITLE
fix(Scaffolder): The filename or extension is too long. #2527

### LINKED ISSUE #2527: Bug (Scaffolder): The filename or extension is too long.
After #2448 fix, getting different error.

docError
error
error2

## Context
Please refer #2448 for more details.

### PR DESCRIPTION
**Pull request by XBeg9**
Fixes #2527, thanks @freben for the help

### CODE REVIEW

**Comment by benjdlambert** [comment]
I think that we need to check if that property exists, as it looks like if the binary fails then it causes the integration tests to fail.

It's because the library is not installed part of the e2e tests.

Also will need a changeset too :)

File: plugins/scaffolder-backend/src/service/router.ts [hunk 1]
```diff
@@ -30,6 +30,11 @@ import {
 } from '../scaffolder';
 import { CatalogEntityClient } from '../lib/catalog';
 import { validate, ValidatorResult } from 'jsonschema';
+import NodeGit from 'nodegit';
+
+// Enables core.longpaths on windows to prevent crashing when checking out repos with long foldernames and/or deep nesting
+// @ts-ignore
+NodeGit.Libgit2.opts(28, 1);
```
Comment (OrkoHunter) [inline plugins/scaffolder-backend/src/service/router.ts:37]: Let's do this to fix the failing tests. (As Ben suggested above)

```ts
NodeGit?.Libgit2?.opts(28, 1);
```

Somehow jest is not able to locate these type definitions. I had tried doing `import * as Nodegit from 'nodegit'` based on this [Stackoverflow Q&A](https://stackoverflow.com/questions/50427196/jest-typeerror-cannot-read-property-of-undefined), but then the error changes and it's not able to find `.opts`.

Comment (XBeg9) [reply plugins/scaffolder-backend/src/service/router.ts]: Thanks, @OrkoHunter ;)

**Comment by XBeg9** [comment]
I've updated and rebased my branch on top of latest `master`. Not sure what's wrong again with CI.

**Comment by XBeg9** [comment]
Is master broken? 
```
[Check Failed] package.json has mismatched field, dependencies[@backstage/core], change from "^0.4.1" to "^0.4.2"?
[Check Failed] package.json has mismatched field, devDependencies[@backstage/cli], change from "^0.4.2" to "^0.4.3"?
[Check Failed] package.json has mismatched field, devDependencies[@backstage/test-utils], change from "^0.1.5" to "^0.1.6"?
```

**Comment by OrkoHunter** [comment]
This is coming from the FOSSA plugin which was merged just now. https://github.com/backstage/backstage/pull/3674
Investigating further. Should be required to bump versions since we made a new release just today!

**Comment by Rugvip** [comment]
Thank you! Will close this though, as we've just moved away from using `nodegit` at all in #3805

**Pull request closed without merging**